src/main/java/gov/nysenate/openleg/common/util/MailUtils.java

```diff
@@ -50,44 +56,82 @@ public MailUtils(@Value("${mail.smtp.host}") String host,
         mailProperties.put("mail.smtp.starttls.enable", stlsEnable);
         mailProperties.put("mail.smtp.ssl.enable", sslEnable);
         mailProperties.put("mail.smtp.ssl.protocols", sslProtocol);
-
         mailProperties.put("mail.smtp.user", smtpUser);
 
-        if (StringUtils.isNotBlank(environment.getEmailFromAddress())) {
+        if (!environment.getEmailFromAddress().isBlank())
             mailProperties.put("mail.smtp.from", environment.getEmailFromAddress());
-        }
 
         mailProperties.put("mail.debug", debug);
-
         mailProperties.put("mail.store.protocol", storeProtocol);
         mailProperties.put("mail.imaps.ssl.protocols", imapsSSLProtocol);
         mailProperties.put("mail.smtp.connectiontimeout", connTimeout);
         mailProperties.put("mail.smtp.timeout", smtpTimeout);
         mailProperties.put("mail.smtp.writetimeout", writeTimeout);
+
         this.environment = environment;
     }
 
-
     /**
-     * Gets an authenticated smtp mail session
-     *
-     * @return Session
+     * Creates a new Store and new Folders. Ensures we can reconnect if the connection fails.
      */
-    public Session getSmtpSession() {
-        return Session.getInstance(mailProperties, getSmtpAuthenticator());
+    public void createCheckMailConnection() throws MessagingException {
+        store = null;
```

Comment:
Should we close the previous store before re connecting?
```suggestion
        store.close();
```

Reply:
Good catch, I just pushed this up. Note that the store still needs to be reassigned to null after, in case i doesn't reconnect properly.